Source/WebKit/UIProcess/Extensions/WebExtension.cpp

```diff
@@ -1137,6 +1147,229 @@ void WebExtension::populateContentScriptPropertiesIfNeeded()
     }
 }
 
+#if ENABLE(WK_WEB_EXTENSION_SIDEBAR)
+bool WebExtension::hasSidePanel()
+{
+    return hasRequestedPermission(WebExtensionPermission::sidePanel());
+}
+#endif // ENABLE(WK_WEB_EXTENSION_SIDEBAR)
+
+const WebExtension::PermissionsSet& WebExtension::supportedPermissions()
+{
+    static MainThreadNeverDestroyed<PermissionsSet> permissions = std::initializer_list<String> { WebExtensionPermission::activeTab(), WebExtensionPermission::alarms(), WebExtensionPermission::clipboardWrite(),
+        WebExtensionPermission::contextMenus(), WebExtensionPermission::cookies(), WebExtensionPermission::declarativeNetRequest(), WebExtensionPermission::declarativeNetRequestFeedback(),
+        WebExtensionPermission::declarativeNetRequestWithHostAccess(), WebExtensionPermission::menus(), WebExtensionPermission::nativeMessaging(), WebExtensionPermission::notifications(), WebExtensionPermission::scripting(),
+        WebExtensionPermission::storage(), WebExtensionPermission::tabs(), WebExtensionPermission::unlimitedStorage(), WebExtensionPermission::webNavigation(), WebExtensionPermission::webRequest(),
+#if ENABLE(WK_WEB_EXTENSIONS_SIDEBAR)
+        WebExtensionPermission::sidePanel(),
```

Comment:
You don't declare this in `WebExtensionPermission`.

Reply:
Are you saying I don't declare the WK_WEB_EXTENSIONS_SIDEBAR, or the sidePanel permission?

Reply:
`sidePanel()` permission was missing.